File: src/sub.cpp
//Inspired from: http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29

#include "ros/ros.h"

//Customized message
#include <performance_tests/SuperAwesome.h>

using namespace ros;

int rate = 0;
Time begin;

void Subscribe_function(const performance_tests::SuperAwesome msg) {
	//Count the number of messages received
	rate++;

	//If a second has passed, the number of messages received so far is the rate
	if(Time::now().toSec()-begin.toSec() > 1) {
		std::cout << "Rate: " << rate << std::endl;
		rate = 0;
		begin = Time::now();
	}
}

int main(int argc, char **argv) {
	//Make sure an input is given
	if (argc == 1) {
		std::cout << "Please provide a topic name" << std::endl;
		return -1;
	}

	// Initialize ROS
	init(argc, argv, "Subscriber_cpp");
	NodeHandle n;

	//Create subscriber object
	Subscriber sub = n.subscribe(argv[1], 1000, Subscribe_function);
	begin = Time::now();

	spin();

	return 0;
}

File: scripts/sub.py
#!/usr/bin/env python
#Inspired from: http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28python%29

import rospy, sys
from performance_tests.msg import SuperAwesome

rate = 0
begin = 0

def Subscriber_Function(data):
	global rate
	global begin

	#Count the number of messages received
	rate += 1

	#If a second has passed, the number of messages received so far is the rate
	if rospy.get_rostime().secs-begin > 0:
		print "Rate: ", rate
		rate = 0
		begin = rospy.get_rostime().secs


def listener():
	#Initialize ROS
	rospy.init_node('Subscriber_python', anonymous=True)

	#Subscribe the desired topic
	rospy.Subscriber(sys.argv[1], SuperAwesome, Subscriber_Function)

	#Initialize the begin time parameter
	global begin
	begin = rospy.get_rostime().secs

	rospy.spin()

if __name__ == '__main__':	
	listener()